lib/nerves_hub_web/views/deployment_view.ex

```diff
@@ -77,6 +77,27 @@ defmodule NervesHubWeb.DeploymentView do
     "#{f.version} #{f.uuid}"
   end
 
+  def archive_dropdown_options(acrhives) do
+    acrhives
+    |> Enum.sort_by(
+      fn archive ->
+        case Version.parse(archive.version) do
+          {:ok, version} ->
+            version
+
+          :error ->
+            %Version{major: 0, minor: 0, patch: 0}
+        end
+      end,
+      {:desc, Version}
+    )
+    |> Enum.map(&[value: &1.id, key: archive_display_name(&1)])
+  end
+
+  def archive_display_name(%{} = a) do
+    "#{a.version} #{a.platform} #{a.architecture} #{a.uuid}"
```

Comment:
```suggestion
    "#{a.version} #{a.uuid} #{a.platform} #{a.architecture}"
```

I feel like version and UUID are probably the most important here. Not sure if we even need the rest, but at least have those 2 together and leftmost

Reply:
This was a copy paste from firmware, so we might want to update that one in another PR